tensorflow/python/framework/fast_tensor_util.pyx

```diff
@@ -7,6 +7,18 @@ cimport numpy as np
 from tensorflow.python.util import compat
 
 
+def AppendFloat16ArrayToTensorProto(
+    # For numpy, npy_half is a typedef for npy_uint16,
+    # see: https://github.com/numpy/numpy/blob/master/doc/source/reference/c-api.coremath.rst#half-precision-functions
+    # Because np.float16_t dosen't exist in cython, we use uint16_t here.
+    # TODO: Use np.float16_t when cython supports it.
+    tensor_proto, np.ndarray[np.uint16_t, ndim=1] nparray):
+  cdef long i, n
+  n = nparray.size
+  for i in range(n):
+    tensor_proto.half_val.append(nparray[i])
```

Comment:
Instead of calling .append in a loop, try using the .extend method on tensor_proto.half_val, which can append an entire list with one call.

Reply:
I compare the performance, and find that `append` is better than `extend`. (Also note that float32 became slow, I run the same test on the same working machine):


```bash
original:
float32: 0.4189 sec
float16: 13.7551 sec


append:
1st experiment:
float32: 0.4094 sec
float16: 1.5520 sec

2nd experiment:
float32: 0.4123 sec
float16: 1.5605 sec


extend:
1st experiment:
float32: 0.4117 sec
float16: 1.8706 sec

2nd experiment:
float32: 0.4648 sec
float16: 1.9272 sec
```

Reply:
Thanks for testing.